src/synthetic_dataset_generator/apps/rag.py

```diff
@@ -0,0 +1,918 @@
+                    )
+                )
+        metadata = [
+            rg.IntegerMetadataProperty(
+                name=f"{item}_length", title=f"{item.capitalize()} length"
+            )
+            for item in ["context", "question", "response"]
+        ]
+
+        vectors = [
+            rg.VectorField(
+                name=f"{item}_embeddings",
+                dimensions=get_sentence_embedding_dimensions(),
+            )
+            for item in ["context", "question", "response"]
+        ]
+        settings = rg.Settings(
+            fields=fields,
+            questions=questions,
+            metadata=metadata,
+            vectors=vectors,
+            guidelines="Please review the conversation and provide an evaluation.",
+        )
+
+        dataframe["chat"] = dataframe.apply(
+            lambda row: [
+                {"role": "user", "content": row["question"]},
+                {"role": "assistant", "content": row["response"]},
+            ],
+            axis=1,
+        )
+
+        for item in ["context", "question", "response"]:
+            dataframe[f"{item}_length"] = dataframe[item].apply(len)
+            dataframe[f"{item}_embeddings"] = get_embeddings(dataframe[item].to_list())
+
+        progress(0.5, desc="Creating dataset")
+        rg_dataset = client.datasets(name=repo_name, workspace=hf_user)
+        if rg_dataset is None:
+            rg_dataset = rg.Dataset(
+                name=repo_name,
+                workspace=hf_user,
+                settings=settings,
+                client=client,
+            )
+            rg_dataset = rg_dataset.create()
+
+        progress(0.7, desc="Pushing dataset to Argilla")
+        hf_dataset = Dataset.from_pandas(dataframe)
+        rg_dataset.records.log(records=hf_dataset)
+        progress(1.0, desc="Dataset pushed to Argilla")
+    except Exception as e:
+        raise gr.Error(f"Error pushing dataset to Argilla: {e}")
+    return ""
+
+
+def show_system_prompt_visibility():
+    return {system_prompt: gr.Textbox(visible=True)}
+
+
+def hide_system_prompt_visibility():
+    return {system_prompt: gr.Textbox(visible=False)}
+
+
+def show_document_column_visibility():
+    return {document_column: gr.Dropdown(visible=True)}
+
+
+def hide_document_column_visibility():
+    return {document_column: gr.Dropdown(visible=False)}
+
+
+def show_pipeline_code_visibility():
+    return {pipeline_code_ui: gr.Accordion(visible=True)}
+
+
+def hide_pipeline_code_visibility():
+    return {pipeline_code_ui: gr.Accordion(visible=False)}
+
+
+######################
+# Gradio UI
+######################
+
+
+with gr.Blocks() as app:
+    with gr.Column() as main_ui:
+        gr.Markdown("## 1. Select your input")
+        with gr.Row(equal_height=False):
+            with gr.Column(scale=2):
+                input_type = gr.Dropdown(
+                    label="Input type",
+                    choices=["dataset-input", "file-input", "prompt-input"],
```

Comment:
not 100% sure about the usefulness of the prompt input, perhaps you can try to come up with a blog to show the effectiveness of this approach?

Also the file-input being markdown feels limited. Either include basic things like PDF, Docx, etc, or redirect users to another tool/Space to upload their own data and first create a chunked HF dataset from it.

Let's discuss a bit later with Dani.

Reply:
Did you explore DocX PDF and such?